### DISCUSSION TITLE
What do you use the library for in real life?

### DISCUSSION THREAD

**Discussion by thorewi** [opis/closure#147] [General]
Hi,

I just wanted to ask for a few examples of what you use the library for in real life. Why do you need to serialize closures? I can see some drawbacks:

1) Not everything is possible to serialize, for example Doctrine EntityManager class (because of PDO Connection), which I need in most closures. I can overcome this by passing it manually when calling the closure, but then it's not safe against code changes.
2) There can be a bug in implementation (the library is not trivial).
3) There can be a security issue.

What are the advantages? When do you use it? Thank you very much.

**Comment by iPurpl3x** [comment]
We (at @glowingblue) use it to push job to a queue worker (the jobs are stored in Redis). Those jobs are implemented in a generic way and therefore, to process the data we always need to pass a Closure that describes the part of the processing that cannot be generalized. It's usually a very short and simple function that has to serialized. 

Examples:
```php
$eventCallbacks = [
    'getUserId' => fn ($event) => $event->discussion->user_id,
    'getDate' => fn ($event) => $event->discussion->created_at,
];
```
or 
```php
$queue->push(new AddPendingActivitiesJob(
    null,
    fn () => $eventRecord?->event?->state?->user_id,
    fn () => $eventRecord?->created_at,
    $describerKey,
));
```
